### PR TITLE
Issue-226 ContainsValidator

### PR DESCRIPTION
**Pull request by Asamsig**
I have implemented the ContainsValidator, I based it mostly on the ItemsValidator.
I noticed we already had tests for this, so I simply uncommented these.

Based on the tests, if the validation is attached to something not an array, we simply skip the validation, I guess this is fine?

### CODE REVIEW

File: src/main/java/com/networknt/schema/ContainsValidator.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        if (node.isEmpty()) {
+            // Array was empty
+            return buildErrorMessageSet(at);
+        } else if (node.isArray()) {
+            boolean contains = false;
+            int i = 0;
+            for (JsonNode n : node) {
+                contains = contains || schema.validate(n, rootNode, at + "[" + i + "]").isEmpty();
+                i++;
```
Comment (Asamsig) [inline src/main/java/com/networknt/schema/ContainsValidator.java]: Perhaps adding a `if(contains) break;` would be ideal, to decrease interations.

Comment (stevehu) [reply src/main/java/com/networknt/schema/ContainsValidator.java]: For most of the scenarios, it doesn't matter as the number of elements won't be big. However, we are trying to provide a generic library and we cannot assume the use cases from users. There might be some crazy use cases with too many elements so I think you are right about to add the break.

Comment (Asamsig) [reply src/main/java/com/networknt/schema/ContainsValidator.java]: I have pushed a new version, where I made it short circuit on the first match. This also somewhat simplifies the code.

File: src/main/java/com/networknt/schema/ValidatorTypeCode.java [hunk 2]
```diff
@@ -71,7 +71,8 @@ public JsonValidator newValidator(String schemaPath, JsonNode schemaNode, JsonSc
     EXCLUSIVE_MINIMUM("exclusiveMinimum", "1039", new MessageFormat("{0}: must have a exclusive minimum value of {1}"), ExclusiveMinimumValidator.class, 14),
     TRUE("true", "1040", null, TrueValidator.class, 14),
     FALSE("false", "1041", new MessageFormat("Boolean schema false is not valid"), FalseValidator.class, 14),
-    CONST("const", "1042", new MessageFormat("{0}: must be a constant value {1}"), ConstValidator.class, 14);
+    CONST("const", "1042", new MessageFormat("{0}: must be a constant value {1}"), ConstValidator.class, 14),
+    CONTAINS("contains", "1043", new MessageFormat("{0}: did not contain an element that passes these validations: {1}"), ContainsValidator.class, 14);
```
[the remark this answers is not in the document]
Comment (Asamsig) [reply src/main/java/com/networknt/schema/ValidatorTypeCode.java]: Better to be consistent, I'll update the PR. :)

**Comment by stevehu** [comment]
@Asamsig I have released 1.0.31 with the contains validator. Thanks a lot for your contribution. 

https://github.com/networknt/json-schema-validator/releases/tag/1.0.31

**Pull request merged**